libs/libcommon/src/libcommon/viewer_utils/asset.py

```diff
@@ -67,71 +78,102 @@ class ImageSource(TypedDict):
     width: int
 
 
-def create_image_file(
-    dataset: str,
-    config: str,
-    split: str,
-    row_idx: int,
-    column: str,
-    filename: str,
-    image: Image.Image,
-    assets_base_url: str,
-    assets_directory: StrPath,
-    overwrite: bool = True,
-) -> ImageSource:
-    dir_path, url_dir_path = create_asset_dir(
-        dataset=dataset,
-        config=config,
-        split=split,
-        row_idx=row_idx,
-        column=column,
-        assets_directory=assets_directory,
-    )
-    makedirs(dir_path, ASSET_DIR_MODE, exist_ok=True)
-    file_path = dir_path / filename
-    if overwrite or not file_path.exists():
-        image.save(file_path)
-    return {
-        "src": f"{assets_base_url}/{url_dir_path}/{filename}",
-        "height": image.height,
-        "width": image.width,
-    }
-
-
 class AudioSource(TypedDict):
     src: str
     type: str
 
 
-def create_audio_file(
+SupportedSource = Union[ImageSource, AudioSource]
+
+
+def upload_asset_file(
+    url_dir_path: str,
+    filename: str,
+    file_path: Path,
+    overwrite: bool = True,
+    s3_client: Optional[S3Client] = None,
+    s3_bucket: Optional[str] = None,
+    s3_folder_name: Optional[str] = None,
+) -> None:
+    if s3_client is not None and s3_bucket is not None:
+        object_key = f"{s3_folder_name}/{url_dir_path}/{filename}"
+        create_object = overwrite or not s3_client.exists_in_bucket(s3_bucket, object_key)
```

Comment:
nit: no need for the intermediate variable

Reply:
I removed it